src/main/java/com/facebook/openwifirrm/ucentral/models/AggregatedState.java

```diff
@@ -0,0 +1,239 @@
+			return channel == other.channel &&
+				channelWidth == other.channelWidth && txPower == other.txPower;
+		}
+	}
+
+	public String bssid;
+	public String station;
+	public long connected;
+	public long inactive;
+	public List<Integer> rssi;
+	public long rxBytes;
+	public long rxPackets;
+	public AggregatedRate rxRate;
+	public long txBytes;
+	public long txDuration;
+	public long txFailed;
+	public long txPackets;
+	public AggregatedRate txRate;
+	public long txRetries;
+	public int ackSignal;
+	public int ackSignalAvg;
+	public Radio radio;
+
+	/** Constructor with no args */
+	public AggregatedState() {
+		this.rxRate = new AggregatedRate();
+		this.txRate = new AggregatedRate();
+		this.rssi = new ArrayList<>();
+		this.radio = new Radio();
+	}
+
+	/** Construct from Association and radio */
+	public AggregatedState(
+		Association association,
+		Map<String, Integer> radioInfo
+	) {
+		this.rxRate = new AggregatedRate();
+		this.txRate = new AggregatedRate();
+		this.rssi = new ArrayList<>();
+
+		this.bssid = association.bssid;
+		this.station = association.station;
+		this.connected = association.connected;
+		this.inactive = association.inactive;
+		this.rssi.add(association.rssi);
+		this.rxBytes = association.rx_bytes;
+		this.rxPackets = association.rx_packets;
+		this.rxRate.add(association.rx_rate);
+		this.txBytes = association.tx_bytes;
+		this.txDuration = association.tx_duration;
+		this.txFailed = association.tx_failed;
+		this.txPackets = association.tx_packets;
+		this.txRate.add(association.tx_rate);
+		this.txRetries = association.tx_retries;
+		this.ackSignal = association.ack_signal;
+		this.ackSignalAvg = association.ack_signal_avg;
+		this.radio = new Radio(radioInfo);
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(bssid, station, radio);
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj) {
+			return true;
+		}
+		if (obj == null) {
+			return false;
+		}
+		if (getClass() != obj.getClass()) {
+			return false;
+		}
+
+		AggregatedState other = (AggregatedState) obj;
+
+		return bssid == other.bssid && station == other.station &&
+			connected == other.connected && inactive == other.inactive && rssi
+				.equals(other.rssi) &&
+			rxBytes == other.rxBytes && rxBytes == other.rxPackets &&
+			Objects.equals(rxRate, other.rxRate) &&
+			txBytes == other.txBytes && txDuration == other.txDuration &&
+			txFailed == other.txFailed && txPackets == other.txPackets &&
+			Objects.equals(txRate, other.txRate) &&
+			txRetries == other.txRetries && ackSignal == other.ackSignal &&
+			ackSignalAvg == other.ackSignalAvg &&
+			Objects.equals(radio, other.radio);
+	}
+
+	/**
+	 * Add an AggregatedState to this AggregatedState. Succeed only when the two
+	 * matches in hashCode.
+	 *
+	 * @param state input AggregatedState
+	 * @return boolean true if the two matches in bssid, station, channel,
+	 *         channel_width and tx_power
+	 */
+	public boolean add(AggregatedState state) {
```

Comment:
You are right.

Reply:
Since you're planning to do a follow-up PR to store time stamps, you could resolve this in that follow-up, right? Since time stamps would make this problem irrelevant?

Reply:
Ok, looks like you just removed all the assignments that overrode the old values. That works.